src/Employer/Employer.Web/Caching/Cache.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Esfa.Recruit.Employer.Web.Caching
+{
+    public class Cache : ICache
+    {
+        private readonly IMemoryCache _memoryCache;
+
+        public Cache(IMemoryCache memoryCache)
+        {
+            _memoryCache = memoryCache;
+        }
+
+        public Task<T> CacheAsideAsync<T>(string key, DateTime absoluteExpiration, Func<Task<T>> objectToCache)
```

Comment:
`async` all the way?

```
public async Task<T> CacheAsideAsync<T>(string key, DateTime absoluteExpiration, Func<Task<T>> objectToCache)
        {
            return await _memoryCache.GetOrCreateAsync(key, entry =>
            {
                entry.AbsoluteExpiration = absoluteExpiration;

                return objectToCache();
            });
        }
```

Reply:
👍 I'll use GetOrCreateAsync. But as I'm not doing anything with the result I can just pass through the task so I don't need to add await/async.